products/statement-generator/src/tests/buttons.test.tsx

```diff
@@ -0,0 +1,144 @@
+    );
+
+    const button = getByText(/Click Me/i);
+
+    fireEvent.click(button);
+
+    expect(mockOnClick).toHaveBeenCalledTimes(1);
+  });
+
+  // Test that buttons render arrows based on the hasBackArrow and hasForwardArrow props
+  test('Button renders back arrow', () => {
+    const { getByText, getByTestId } = render(
+      <ButtonComponent buttonText="Back" hasBackArrow />
+    );
+
+    const button = getByText(/Back/i);
+    const arrow = getByTestId('back-arrow');
+
+    expect(button).toBeTruthy();
+    expect(arrow).toBeTruthy();
+  });
+
+  test('Button renders forward arrow', () => {
+    const { getByText, getByTestId } = render(
+      <ButtonComponent buttonText="Next" hasForwardArrow />
+    );
+
+    const button = getByText(/Next/i);
+    const arrow = getByTestId('forward-arrow');
+
+    expect(button).toBeTruthy();
+    expect(arrow).toBeTruthy();
+  });
+
+  // Test that the button is disabled when the disabled prop is true
+  test('Button is disabled when disabled prop is true', () => {
+    const { getByRole } = render(
+      <ButtonComponent buttonText="Disabled Button" disabled={true} />
+    );
+
+    const button = getByRole('button', { name: /Disabled Button/i });
+
+    if (button) {
+      expect(button).toBeDisabled();
+    }
+  });
+
+  // Test that the button is not disabled when the disabled prop is false
+  test('Button is re-enabled when disabled prop is false', () => {
+    const { getByRole } = render(
+      <ButtonComponent buttonText="Enabled Button" disabled={false} />
+    );
+
+    const button = getByRole('button', { name: /Enabled Button/i });
+
+    if (button) {
+      expect(button).not.toBeDisabled();
+    }
+  });
+
+  // Test that the button has appropriate styles when disabled
+  test('Disabled button has appropriate styles', () => {
+    const { getByRole } = render(
+      <ButtonComponent buttonText="Disabled Button" disabled={true} />
+    );
+
+    const button = getByRole('button', { name: /Disabled Button/i });
+
+    expect(button).toHaveStyle({ color: 'rgba(0, 0, 0, 0.26)' });
+    expect(button).toHaveStyle({ background: 'rgb(197, 179, 209)' });
+  });
+
+  // Test that the button has appropriate styles when not disabled
+  test('Non-disabled button has appropriate styles', () => {
+    const { getByRole } = render(
+      <ButtonComponent buttonText="Enabled Button" disabled={false} />
+    );
+
+    const button = getByRole('button', { name: /Enabled Button/i });
+
+    expect(button).not.toHaveStyle({ color: 'rgba(0, 0, 0, 0.26)' });
+    expect(button).not.toHaveStyle({ background: 'rgb(197, 179, 209)' });
+  });
+
+  // Test that the button is at least 44px wide
+  test('Button is at least 44px', () => {
+    const { getByRole } = render(<ButtonComponent buttonText="Click me" />);
+
+    const button = getByRole('button', { name: /Click me/i });
+
+    expect(button).toHaveStyle({ minHeight: '44px' });
```

Comment:
I think this should be minWidth

Reply:
@sydneywalcoff There isn't a width or minWidth specified on the Button component, so testing for it returns false. I could add a minWidth to the actual component if that is what we need though!

Reply:
@whitneywind Great insight! We just want to change _what_ we're testing. The goal of this test is accessibility based and makes sure that the button will be wide enough for folx with low vision. 

I think we want to evaluate the _rendered button's size_ to make sure it's at least 44px.

Reply:
Right, okay I'm having trouble with it because all of its dimensions are coming up as 0. I think it's because the test rendered button is only attached to jest's testing DOM tree. I've been researching, but I'm struggling to find a way to text the actual rendered dimensions using jest.